src/platform/macos/view.rs

```diff
@@ -412,15 +450,16 @@ extern fn key_down(this: &Object, _sel: Sel, event: id) {
 
         state.raw_characters = get_characters(event);
 
-        let keycode: c_ushort = msg_send![event, keyCode];
         // We are checking here for F21-F24 keys, since their keycode
         // can vary, but we know that they are encoded
         // in characters property.
+        let keycode = to_translated_macos_scancode(event);
         let virtual_keycode = to_virtual_key_code(keycode)
             .or_else(|| {
                 check_additional_virtual_key_codes(&state.raw_characters)
             });
-        let scancode = keycode as u32;
+
+        let scancode: u32 = msg_send![event, keyCode];
```

Comment:
I think it would be good to put all the `keyCode` calls into a function called `get_scancode` or something, and then have comments explicitly noting that `keyCode` returns the scancode. That way, there won't be much room for confusion in the future (the reason this bug never got caught is largely due to how AppKit's usage of the label `keyCode` differs from ours).